scipy/signal/signaltools.py

```diff
@@ -458,16 +512,32 @@ def convolve(in1, in2, mode='full'):
     if volume.ndim == kernel.ndim == 0:
         return volume * kernel
 
+    out_shape = {'full':[n + k for n, k in zip(volume.shape, kernel.shape)],
+                 'same':volume.shape,
+                 'valid':[n - k + 1 for n, k in zip(volume.shape,
+                                                    kernel.shape)]}
+
+    # see whether the fourier transform convolution method or the direct
+    # convolution method is faster (discussed in scikit-image PR #1792)
+    big_O_constant = 5e-6 if volume.ndim > 1 else 4.81e-4
+    direct_time = big_O_constant * np.prod(volume.shape + kernel.shape +
```

Comment:
`prod(volume.shape + kernel.shape + tuple(out_shape[mode]))` can be simplified to

`volume.size * kernel.size * prod(out_shape[mode])`